test/view/rendering/null_local_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+
+describe Hanami::View::Rendering::NullLocal do
+  before do
+    @null = Hanami::View::Rendering::NullLocal.new(:result)
+  end
+
+  it 'does not complain for unknown sent messages' do
+    actual = @null.foo
+
+    actual.must_be_instance_of(Hanami::View::Rendering::NullLocal)
+    actual.inspect.must_match ":result.foo"
+  end
+
+  it 'returns empty string for to_str' do
+    @null.to_str.must_equal ''
+  end
+
+  it 'returns false to all?' do
+    @null.all?.must_equal false
+  end
+
+  it 'returns false to any?' do
+    @null.any?.must_equal false
+  end
+
+  it 'returns true to empty?' do
+    @null.empty?.must_equal true
+  end
+
+  it 'returns true to nil?' do
+    @null.nil?.must_equal true
```

Comment:
@davydovanton `NullLocal` inherits from Ruby's `BasicObject`.

In the past I got odd testing results because Minitest assumes that the current object has a set of features that the descendants of `Object` have.

The difference here is subtle, but intended.

With this syntax, the object under test is `@null`. But we just said that its nature may lead to unexpected testing results.

```ruby
@null.must_be_nil
```

If you look closer, you're asking if `@null` is `nil`, but does it? NO.

We want to test if `@null`, respond to `#nil?` and returns `true`.

```ruby
@null.nil?.must_equal true
```

In this case, the object under test is the returning value of `#nil?`: a boolean, which is safer to test.